Docs maintenance (#429)
PixelgenTechnologies/pixelator: docs/README.md

  - A logging filter (`_keep_warning`) drops a small set of warnings that are emitted without a Sphinx type/subtype and so cannot be matched by `suppress_warnings`.
- When new code introduces a cross-reference warning, prefer fixing the reference.

## Maintenance

- Update or add docstrings and CLI help-text in the source code as the code changes.
- The overview page (`overview.rst`) lists some primary entry points of the API. Additions to this page are made manually.
- `intersphinx_mapping` in `docs/conf.py` lists the URLs for documentation of external names used for cross-referencing. This list is maintained manually.
- Keep this handbook (`docs/README.md`) up to date when the docs setup changes.

## Deployment

- Building and deploying the docs is handled by the "Docs" workflow (`.github/workflows/deploy-docs.yml`).
- The workflow runs on three triggers: published releases, manual dispatch (`workflow_dispatch`, which requires a `tag` input), and pull requests.
- The build job runs for all three triggers and builds with `sphinx-build ... --fail-on-warning -n`, so warnings fail the build (see "Warnings and strict builds").

> - The deploy job only runs for manual dispatch and non-prerelease releases. It does not run on pull requests, and prereleases are built but not deployed.

Reviewer: ```suggestion
- The deploy job only runs for manual dispatch and stable releases. It does not run on pull requests nor prereleases.
```
